src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,50 @@
+import {Directive, EventEmitter, HostListener, Input, Output} from '@angular/core';
+
+export enum LoadDirection {
+    none = 0,
+    top,
+    bottom,
+}
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    private oldScrollTop = 0;
+    private oldLoadDirection = LoadDirection.none;
+
+    @Input() loadDataBorderOffset = 100;
+
+    @Output() loadData = new EventEmitter<LoadDirection>();
+
+    @HostListener('scroll', ['$event.target'])
+    onScroll(target: HTMLElement) {
+        const {scrollTop, clientHeight, scrollHeight} = target;
```

Comment:
Рекомендую в таких случаях использовать деструктуризацию в аргументах:
```suggestion
    onScroll({scrollTop, clientHeight, scrollHeight}: HTMLElement) {
```

Reply:
Принял, исправил